src/ripple/ledger/ApplyView.h

```diff
@@ -273,22 +273,19 @@ class ApplyView : public ReadView
               allowable pages.
     */
     /** @{ */
-    std::optional<std::uint64_t>
-    dirAppend(
-        Keylet const& directory,
-        uint256 const& key,
-        std::function<void(std::shared_ptr<SLE> const&)> const& describe)
-    {
-        return dirAdd(true, directory, key, describe);
-    }
-
     std::optional<std::uint64_t>
     dirAppend(
         Keylet const& directory,
         Keylet const& key,
         std::function<void(std::shared_ptr<SLE> const&)> const& describe)
     {
-        return dirAppend(directory, key.key, describe);
+        if (key.type != ltOFFER)
+        {
+            assert(!"Only Offers are appended to book directories.  "
+                "Call dirInsert() instead.");
+            return std::nullopt;
+        }
+        return dirAdd(true, directory, key.key, describe);
```

Comment:
The "High Level Overview of Change" in the PR through me off at first since it says "remove the calls to dirAdd()". Perhaps it should clarify what "dirAdd()" should be deleted?

Reply:
I appreciate the `assert` to try to keep us from doing anything weird, but I'm not sure it's a good idea. This function is intended to append to the particular directory, instead of trying to fit it into the first page with an empty slot. It's true that this operation is currently only used by order books (so as to maintain temporal ordering of offers), but there's nothing that prevents this from being used differently, so I think the `assert` my communicate the wrong information.

For example, imagine if we added a new directory for, say, NFT offers.

Reply:
At @nbougalis, I think the `assert` is a good idea because it helps keep folks less familiar with the code on the straight and narrow.  To someone who is not intimate with this code, it may not be obvious whether a particular object should be added or appended to the directory.  As it stands there's currently only one situation where append is appropriate: offers in book directories.  All other cases use add.

In the future there may, indeed, be other situations where appending is the right thing to do.  But the source code is malleable, so the assert can be removed or changed when the new situation arises.

Just my take...